java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -192,110 +188,129 @@ private static byte[] buildMessage(
     return builder.sizedByteArray();
   }
 
-  public List<String> takeState(String clientFeatures) throws YggdrasilInvalidInputException {
+  public void takeState(String clientFeatures) throws YggdrasilInvalidInputException {
     try {
       customStrategiesEvaluator.loadStrategiesFor(clientFeatures);
       byte[] messageBytes = clientFeatures.getBytes(StandardCharsets.UTF_8);
       nativeInterface.takeState(this.enginePointer, messageBytes);
-    } catch (TrapException e) {
-      throw e;
+    } catch (RuntimeException e) {
+      throw new YggdrasilInvalidInputException(
+          "Failed to take state, ensure the input is a valid Yggdrasil state file", e);
```

Comment:
Changing this here because this isn't how format exceptions get bubbled. This should only occur if something goes wrong deep in Chicory
```suggestion
          "Failed to take state:", e);
```